src/plugins/autocompletion_providers/Vagrant.ts

```diff
@@ -0,0 +1,40 @@
+import {PluginManager} from "../../PluginManager";
+import {linedOutputOf} from "../../PTY";
+import {styles, Suggestion, contextIndependent} from "./Common";
+
+const commands = contextIndependent(async() => {
+    let lines: string[] = [];
+
+    try {
+        lines = await linedOutputOf("vagrant", ["list-commands"], process.env.HOME);
+    } catch (e) {
+        // skip if command is not exist
+    }
+
+    return lines
+        .map(line => {
+            const matches = line.match(/([\-a-zA-Z0-9]+)  /);
+
+            if (matches) {
+                const name = matches[1];
+                const description = line.replace(matches[1], "").trim();
+
+                return new Suggestion()
```

Comment:
Please use the new API, where you'd pass everything into the constructor. You can find examples in any other provider.